### PR TITLE
feat: add DataTable display and sort methods

### LINKED ISSUE #282: Add custom DataTable sorting method
The current DataTable sorting just works fine. It would also be great to be able to set a custom sorting method to manage situations where the displayed data is treated differently than the ‘raw’ data, e.g. currency, should be sorted as a number (`3217.58`) and displayed as a formatted string (`$3,217.58`).

### PR DESCRIPTION
**Pull request by albertms10**
fixes #282

---

- feat: add display and sort methods in header object
- feat: update types and Component Index
- feat: update Public API with datatable headers methods

### CODE REVIEW

File: src/DataTable/DataTable.svelte [hunk 1]
```diff
@@ -231,7 +241,9 @@
                 dispatch('click:cell', cell);
               }}"
             >
-              <slot name="cell" row="{row}" cell="{cell}">{cell.value}</slot>
+              <slot name="cell" row="{row}" cell="{cell}">
+                {headers[j].display ? headers[j].display(cell.value) : cell.value}
```
Comment (albertms10) [inline src/DataTable/DataTable.svelte:245]: Should the `display` method be accessed this way—referencing the original `headers` array—or by passing it to the cells object?

https://github.com/IBM/carbon-components-svelte/blob/ead9634ab641ce50b13046604687164ee86be8cd/src/DataTable/DataTable.svelte#L119-L122

File: src/DataTable/DataTable.svelte [hunk 2]
```diff
@@ -85,7 +85,12 @@
   };
   const dispatch = createEventDispatcher();
   const tableSortable = writable(sortable);
-  const sortHeader = writable({ id: null, key: null, sortDirection: "none" });
+  const sortHeader = writable({
+    id: null,
+    key: null,
+    sort: undefined,
```
Comment (albertms10) [inline src/DataTable/DataTable.svelte:91]: Should `sort` be initialized with `null` as the other `sortHeader` properties?

File: src/DataTable/DataTable.svelte (hunk 1 above)
Comment (metonym) [reply src/DataTable/DataTable.svelte:245]: What you have is good

**Review by metonym — APPROVED** [review]
Nice work! This feature makes the `DataTable` very flexible.

Now, you'd be able to:

- customize display/sorting per header as needed
- customize cell value with markup/components using named slots

Can't wait to start using this.

**Pull request merged**